src/mapping_navigation/scripts/pure_pursuit.py

```diff
@@ -18,27 +18,30 @@ class State:
     The yaw of the car
     The cars wheel base
     """
-    def __init__(self, wheel_base: float, x: float=0.0, y:float=0.0, yaw:float=0.0, velocity:float=0.0):
+    def __init__(self, wheel_base: float=2.2, point: Point=Point((0,0)), yaw:float=0.0, velocity:float=0.0):
         self.wheel_base = wheel_base
-        self.x = x
-        self.y = y
+        self.point = point
         self.yaw = yaw
         self.velocity = velocity
-        self.rear_x = self.x - ((wheel_base / 2) * math.cos(self.yaw))
-        self.rear_y = self.y - ((wheel_base / 2) * math.sin(self.yaw))
+        self.rear_x = self.point.coordinate[0] - ((wheel_base / 2) * math.cos(self.yaw))
+        self.rear_y = self.point.coordinate[1] - ((wheel_base / 2) * math.sin(self.yaw))
     
     """
     Updates the cars position
     Cartesian coordinates for the cars position
     Quaterion to find the yaw of the car
     """
-    def update(self, x: float, y: float, quaterion: Tuple[float, float, float, float], velocity:float):
-        self.x = x
-        self.y = y
-        self.velocity = velocity
+    def update_pos(self, point: Point, quaterion: Tuple[float, float, float, float]):
+        self.point = point
         self.yaw = euler_from_quaternion(quaterion)
```

Comment:
Forgot to ask before, why are we using quaterions? I'm aware of gimbal lock, though didn't think it would apply when only yaw is being changed. Could be wrong though.

Reply:
This is more to do with how its usually done, in most instances you return all the euler angles from the quaternion, atm we only need yaw, but I thought I would just send in the whole thing if we needed the other two angles down the line, a bit silly in retrospect so ill remove the unnecessary inputs

Reply:
actually scratch that, had a mental fart. I used quaternion cause of this https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles, I made it input quaternion more for naming convention. Not sure if you still think it warrents a change, only difference is it passing in x,y,z,w vs the tuple

Reply:
So are the units in euler coordinates or quaternions? I'm assuming it's quarternions due to think link (which also mentions same wiki page you did): https://github.com/Microsoft/AirSim/issues/1369 .  If so, in the handle_gps_data in navigation, consider adding a comment saying oritentation is provided as a quaternion.

Reply:
Sorry a bit confused by your question, this method takes in the quaternion and just return the yaw, one of the euler angles, I can see the naming being unclear atm so I will change it to yaw from quaternion